nixos/modules/tasks/filesystems.nix

```diff
@@ -28,10 +29,10 @@ let
   coreFileSystemOpts = { name, config, ... }: {
 
     options = {
-
       mountPoint = mkOption {
         example = "/mnt/usb";
-        type = nonEmptyStr;
+        type = addCheckDesc "non-empty without trailing slash" types.str
+          (s: isNonEmpty s && (builtins.match ".+/$" s) == null);
```

Comment:
nit: I don't think the $ is necessary, since `builtins.match` already requires the whole string to match.